### PR TITLE
Tweaked engulfing sound volume and handling

### PR DESCRIPTION
**Pull request by Kasterisk**
**Brief Description of What This PR Does**

Reduced engulfing sound volume to 0.6 and the sound now fades out smoothly when stopping engulf mode.

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

File: src/microbe_stage/Microbe.cs [hunk 1]
```diff
@@ -712,4 +720,113 @@ private void CountHexes()
...
+    /// <summary>
+    ///   Provides extra level of abstraction to allow simultaneous switching between 3D positional and non positional
+    ///   audio players in a single Spatial-derived class.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Useful in cases where the playing of an audio stream must be able to fulfill both of these conditions.
+    ///   </para>
+    /// </remarks>
+    private class AudioPlayer : Spatial
```
Comment (hhyyrylainen) [inline src/microbe_stage/Microbe.cs:733]: This should probably be in a new file in the engine folder, as I presume for 3D character stages we might have the same issue as currently.

File: src/general/HybridAudioPlayer.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    private AudioStreamPlayer3D player3D;
+    private AudioStreamPlayer playerNonPositional;
+
+    private float volume;
+
+    public HybridAudioPlayer()
+    {
+        player3D = new AudioStreamPlayer3D();
+        playerNonPositional = new AudioStreamPlayer();
+
```
Comment (hhyyrylainen) [inline src/general/HybridAudioPlayer.cs:23]: Would it be completely crazy to suggest that these would be created on demand? With this approach we create potentially hundreds of unnecessary audio player objects.

File: src/microbe_stage/Microbe.Contact.cs [hunk 3]
```diff
@@ -782,13 +782,29 @@ private void HandleEngulfing(float delta)
             if (!engulfAudio.Playing)
                 engulfAudio.Play();
 
+            // To balance loudness, here the engulfment audio's max volume is reduced to 0.6 in linear energy
```
Comment (hhyyrylainen) [inline src/microbe_stage/Microbe.Contact.cs:785]: `linear volume`?

File: src/microbe_stage/Microbe.cs [hunk 4]
```diff
@@ -179,12 +179,26 @@ public override void _Ready()
...
         cellBurstEffectScene = GD.Load<PackedScene>("res://src/microbe_stage/particles/CellBurstEffect.tscn");
 
+        engulfAudio = new HybridAudioPlayer
+        {
+            Stream = GD.Load<AudioStream>("res://assets/sounds/soundeffects/engulfment.ogg"),
+            Positional = !IsPlayerMicrobe,
+            Bus = "SFX",
+        };
+
+        AddChild(engulfAudio);
```
Comment (hhyyrylainen) [inline src/microbe_stage/Microbe.cs:194]: engulfAudio could use _Ready to determine which player type to create as that should trigger here after the properties have been set.

File: src/general/HybridAudioPlayer.cs (hunk 2 above)
Comment (Kasterisk) [reply src/general/HybridAudioPlayer.cs:23]: No that's perfectly sensible, I think Positional property should be readonly.

Comment (Kasterisk) [reply src/general/HybridAudioPlayer.cs]: Added.

**Review by RacerBG — APPROVED** [review]
Sounds good to me while testing it.

File: src/general/HybridAudioPlayer.cs [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+///   audio players in a single Node-derived class. Though setting this can only be done in instantiation.
+/// </summary>
+/// <remarks>
+///   <para>
+///     Useful in cases where the playing of an audio stream must be able to fulfill both of these conditions.
+///   </para>
+/// </remarks>
+public class HybridAudioPlayer : Node
+{
+    public readonly bool Positional;
```
Comment (hhyyrylainen) [inline src/general/HybridAudioPlayer.cs:14]: I'd make this a get only property

Comment (hhyyrylainen) [reply src/general/HybridAudioPlayer.cs:14]: Code looks good except this one thing.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
I'd like to see one small change that shouldn't impact functionality at all, so if that is fixed by tomorrow morning (my time), this can be included in 0.5.6.1

File: src/general/HybridAudioPlayer.cs (hunk 5 above)
Comment (Kasterisk) [reply src/general/HybridAudioPlayer.cs]: Changed.

**Review by hhyyrylainen — APPROVED** [review]
Code looks good now.

**Pull request merged**